lib/salus/scanners/osv/gradle_osv.rb

```diff
@@ -0,0 +1,159 @@
+require 'salus/scanners/osv/base'
+
+module Salus::Scanners::OSV
+  class GradleOSV < Base
+    class SemVersion < Gem::Version; end
+
+    EMPTY_STRING = "".freeze
+    DEFAULT_SOURCE = "https://osv.dev/list".freeze
+    DEFAULT_SEVERITY = "MODERATE".freeze
+    GRADLE_OSV_ADVISORY_URL = "https://osv-vulnerabilities.storage.googleapis.com"\
+        "/Maven/all.zip".freeze
+
+    def should_run?
+      @repository.build_gradle_present?
+    end
+
+    def run
+      # Find dependencies
+      dependencies = find_dependencies
+      if dependencies.empty?
+        err_msg = "GradleOSV: Failed to parse any dependencies from the project."
+        report_stderr(err_msg)
+        report_error(err_msg)
+        return
+      end
+
+      # Fetch vulnerabilities
+      @osv_vulnerabilities ||= fetch_vulnerabilities(GRADLE_OSV_ADVISORY_URL)
+      if @osv_vulnerabilities.nil?
+        err_msg = "GradleOSV: No vulnerabilities found to compare."
+        report_stderr(err_msg)
+        report_error(err_msg)
+        return
+      end
+
+      # Match and Report scanner status
+      vulnerabilities_found = match_vulnerable_dependencies(dependencies)
+      results = group_vulnerable_dependencies(vulnerabilities_found)
+      return report_success if results.empty?
+
+      report_failure
+      log(JSON.pretty_generate(results))
```

Comment:
Did you intended to retain this loggin?

Reply:
Yes that's the one printing to console with the results

Reply:
We should really wrap these, but that can be a salus-wide change for another day